src/physics/em/LivermorePEModel.hh

```diff
@@ -54,12 +47,24 @@ class LivermorePEModel final : public Model
     std::string label() const final { return "Livermore photoelectric"; }
 
     // Access data on device
-    detail::LivermorePEPointers device_pointers() const;
+    inline const detail::LivermorePEPointers& device_pointers() const;
 
   private:
     detail::LivermorePEPointers     interface_;
-    SPConstSubshellIdAllocatorStore vacancies_;
+    detail::RelaxationScratchData<Ownership::value, MemSpace::device>
+        relax_scratch_;
```

Comment:
This means there will be a separate vacancy stack allocator for each model that performs atomic relaxation, correct?

Reply:
That's correct and a good point. For now maybe we should only consider the case of the single (implemented) model, but when we extend to multiple models that use atomic relaxation it does make more sense to share the temporary space between them if we're at all tight on space.